core/amber/src/main/scala/edu/uci/ics/amber/engine/architecture/worker/neo/promisehandlers/PauseHandler.scala

```diff
@@ -1,40 +1,33 @@
 package edu.uci.ics.amber.engine.architecture.worker.neo.promisehandlers
 
 import akka.actor.ActorContext
+import com.twitter.util.Future
 import edu.uci.ics.amber.engine.architecture.worker.neo.WorkerInternalQueue.DummyInput
-import edu.uci.ics.amber.engine.architecture.worker.neo.{WorkerPromiseHandlerInitializer}
+import edu.uci.ics.amber.engine.architecture.worker.neo.WorkerRPCHandlerInitializer
 import edu.uci.ics.amber.engine.architecture.worker.neo.promisehandlers.PauseHandler.WorkerPause
 import edu.uci.ics.amber.engine.common.ambermessage.WorkerMessage.{ExecutionPaused, ReportState}
-import edu.uci.ics.amber.engine.common.promise.{
-  ControlCommand,
-  PromiseCompleted,
-  SynchronizedInvocation
-}
-import edu.uci.ics.amber.engine.common.statetransition.WorkerStateManager._
+import edu.uci.ics.amber.engine.common.promise.RPCServer.RPCCommand
 
 object PauseHandler {
-  final case class WorkerPause() extends ControlCommand[PromiseCompleted]
+  final case class WorkerPause() extends RPCCommand[ExecutionPaused]
 }
 
 trait PauseHandler {
-  this: WorkerPromiseHandlerInitializer =>
+  this: WorkerRPCHandlerInitializer =>
 
-  registerHandler {
-    case WorkerPause() =>
-      // workerStateManager.shouldBe(Running, Ready)
-      val (p, ctx) = createPromise[ExecutionPaused]()
-      pauseManager.registerNotifyContext(ctx)
-      pauseManager.pause()
-      // workerStateManager.transitTo(Pausing)
-      // if dp thread is blocking on waiting for input tuples:
-      if (dataProcessor.isQueueEmpty) {
-        // insert dummy batch to unblock dp thread
-        dataProcessor.appendElement(DummyInput())
-      }
-      p.map { res =>
-        println("pause actually returned")
-        //workerStateManager.transitTo(Paused)
-        returning()
-      }
+  registerHandler { pause: WorkerPause =>
+    // workerStateManager.shouldBe(Running, Ready)
+    val p = pauseManager.pause()
+    // workerStateManager.transitTo(Pausing)
+    // if dp thread is blocking on waiting for input tuples:
+    if (dataProcessor.isQueueEmpty) {
+      // insert dummy batch to unblock dp thread
+      dataProcessor.appendElement(DummyInput())
+    }
+    p.map { res =>
+      println("pause actually returned")
```

Comment:
let's add DP thread may not have paused. Also please use WorkflowLogger

Reply:
When `p.map` happens, the DP thread is paused. So the log statement is correct.

Reply:
Yes.. you are right... The only change is to add a WorkflowLogger